src/Helper/ServerUrl.php

```diff
@@ -85,6 +85,16 @@ protected function detectHost()
                     $this->setHost(substr($_SERVER['HTTP_HOST'], 0, 0-strlen($portStr)));
                     return;
                 }
+
+                // If the Host header contains a port, and it differs from the
+                // SERVER_PORT, use the port from the Host header. Typically
+                // this is a situation where port-forwarding is in use, and you
+                // want to present a URI using the public port.
```

Comment:
> or similar as we don't actually check that it differs from SERVER_PORT (or need to).

There's a block immediately before it that *does*, actually. :) This is the condition that we hit if the SERVER_PORT and the HTTP_HOST's port are not equal.